c_common/front_end_common_lib/include/recording.h

```diff
@@ -153,7 +155,7 @@ void recording_finalise(void);
 //!            a channel is enabled for recording
 //! \return True if the initialisation was successful, false otherwise
 bool recording_initialize(
-        address_t recording_data_address, uint32_t *recording_flags);
+        void **recording_data_address, uint32_t *recording_flags);
```

Comment:
changing void ** to a recording_data_t* and the places around would be a better solution. unfortunately it requires a bit of thought. bounced back at @rowleya

Reply:
OK, so I just went to change this and it reminded me why I didn't do this the first time.  The issue is that the data required is not just a recording_data_t, but also some other things (see the lines below).  So the return is *not* an address to a recording_data_t, so it might be a bit weird to have this type!

Reply:
so what ya saying is that were doing really shite things here. and so we need to keep it as a void **